src/WebJobs.Script/Diagnostics/CategoryTraceWriterFactory.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.IO;
+using Microsoft.Azure.WebJobs.Host;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    internal class CategoryTraceWriterFactory
+    {
+        private readonly string _category;
+        private readonly ScriptHostConfiguration _scriptHostConfig;
+
+        public CategoryTraceWriterFactory(string category, ScriptHostConfiguration scriptHostConfig)
```

Comment:
I like the idea. We can also introduce something into the existing factory (so a new one wouldn't be needed) to output the category.